kahuna/public/js/util/async.js

```diff
@@ -6,94 +6,107 @@ export var async = angular.module('util.async', []);
  * Return a lazy function that will yield before calling the input `func`.
  */
 async.factory('nextTick',
-              ['$timeout',
-               function($timeout) {
+  ['$timeout',
+    function($timeout) {
 
-    function nextTick(func) {
+      function nextTick(func) {
         return () => $timeout(func, 0);
-    }
+      }
 
-    return nextTick;
-}]);
+      return nextTick;
+    }]);
 
 
 /**
  * Return a Promise that is resolved with no value after `duration`.
  */
 async.factory('delay',
-              ['$q', '$timeout',
-               function($q, $timeout) {
+  ['$q', '$timeout',
+    function($q, $timeout) {
 
-    function delay(duration) {
+      function delay(duration) {
         var defer = $q.defer();
         $timeout(defer.resolve, duration);
         return defer.promise;
-    }
+      }
 
-    return delay;
-}]);
+      return delay;
+    }]);
 
 
 async.factory('race',
-              ['$q',
-               function($q) {
+  ['$q',
+    function($q) {
 
-    function race(promises) {
+      function race(promises) {
         var first = $q.defer();
         promises.forEach(promise => {
-            promise.then(first.resolve, first.reject);
+          promise.then(first.resolve, first.reject);
         });
         return first.promise;
-    }
+      }
 
-    return race;
-}]);
+      return race;
+    }]);
 
 
 async.factory('poll',
-              ['$q', 'delay', 'race',
-               function($q, delay, race) {
+  ['$q', 'delay', 'race',
+    function($q, delay, race) {
 
-    function poll(func, pollEvery, maxWait) {
+      function poll(func, pollEvery, maxWait) {
         var timeout = delay(maxWait).then(() => $q.reject(new Error('timeout')));
 
         // Returns the result of promise or a rejected timeout
         // promise, whichever happens first
         function withTimeout(promise) {
-            return race([promise, timeout]);
+          return race([promise, timeout]);
         }
 
+
+        const withBackoff = (()=>{
```

Comment:
Does this need to be an IIFE?

Reply:
can probably bring the closed over vars into parent scope